File: python/455_Assign-Cookies.py
class Solution:
    def findContentChildren(self, g, s):
        g.sort(reverse=True)
        s.sort()
        counter = 0
        
        for cookie in s:
            if not g:
                return counter
            
            if cookie >= g[-1]:
                counter+=1
                g.pop()
        return counter

def main():
    sol = Solution()
    
    g = [1,2,3]
    s = [1,1]
    #Output: 1
    print(sol.findContentChildren(g, s))
    
    g = [1,2]
    s = [1,2,3]
    #Output: 2
    print(sol.findContentChildren(g, s))
    
if __name__ == '__main__':
    main()

File: c++/455_Assign-Cookies.cpp
class Solution {
public:
    int findContentChildren(vector<int>& g, vector<int>& s) {
        sort(g.rbegin(), g.rend());
        sort(s.rbegin(), s.rend());

        int n = g.size(), m = s.size(), cnt = 0;
        for(int i = 0, j = 0; i < n && j < m; ++i){
            if(s[j] >= g[i]){
                ++j;
                ++cnt;
            }
        }
        return cnt;
    }
};

class Solution {
public:
    int findContentChildren(vector<int>& g, vector<int>& s) {
        sort(g.begin(), g.end());
        sort(s.begin(), s.end());

        int gIndex = 0, sIndex = 0, gSize = g.size(), sSize = s.size(), res = 0;
        while(sIndex < sSize && gIndex < gSize){

            if(s[sIndex] >= g[gIndex]){
                ++gIndex;
                ++res;
            }
            ++sIndex;
        }
        return res;  
    } 
};